### PR TITLE
8256052: Remove unused allocation type from fieldInfo

### PR DESCRIPTION
**Pull request by fparain**
Please review this small cleanup code, removing the now unused allocation type from the fieldInfo structure.

Tested with Mach5, tiers 1 to 3 and locally by running test/hotspot/jtreg/serviceability/sa tests.

Thank you,

Fred

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8256052](https://bugs.openjdk.java.net/browse/JDK-8256052): Remove unused allocation type from fieldInfo

### Reviewers
 * [Claes Redestad](https://openjdk.java.net/census#redestad) (@cl4es - **Reviewer**) ⚠️ Review applies to c048832e68f3bc7bb22d43c65fa86f584593f57b
 * [Lois Foltan](https://openjdk.java.net/census#lfoltan) (@lfoltan - **Reviewer**) ⚠️ Review applies to b4b24792e979420061ad6be9f3cab95ad184bc4f
 * [Harold Seigel](https://openjdk.java.net/census#hseigel) (@hseigel - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1130/head:pull/1130`
`$ git checkout pull/1130`

### CODE REVIEW

File: src/hotspot/share/classfile/classFileParser.cpp [hunk 1]
```diff
@@ -1706,7 +1706,6 @@ void ClassFileParser::parse_fields(const ClassFileStream* const cfs,
 
     // Remember how many oops we encountered and compute allocation type
     const FieldAllocationType atype = fac->update(is_static, type);
```
Comment (cl4es) [inline src/hotspot/share/classfile/classFileParser.cpp:1708]: The returned `FieldAllocationType` is never used at either call-site, so maybe the `update` method can be simplified, too? (It seems all `update` does is increment a per-type counter, so the name is a bit surprising)

File: src/hotspot/share/runtime/vmStructs.cpp [hunk 2]
```diff
@@ -2257,8 +2257,8 @@ typedef HashtableEntry<InstanceKlass*, mtClass>  KlassHashtableEntry;
   /*************************************/                                 \
                                                                           \
   declare_preprocessor_constant("FIELDINFO_TAG_SIZE", FIELDINFO_TAG_SIZE) \
-  declare_preprocessor_constant("FIELDINFO_TAG_MASK", FIELDINFO_TAG_MASK) \
   declare_preprocessor_constant("FIELDINFO_TAG_OFFSET", FIELDINFO_TAG_OFFSET) \
+  declare_preprocessor_constant("FIELDINFO_TAG_CONTENDED", FIELDINFO_TAG_CONTENDED) \
```
Comment (cl4es) [inline src/hotspot/share/runtime/vmStructs.cpp:2261]: Not sure it's necessary to add this with no usage?

**Review by cl4es — APPROVED** [review]
Nice cleanup!

File: src/hotspot/share/classfile/classFileParser.cpp (hunk 1 above)
Comment (fparain) [reply src/hotspot/share/classfile/classFileParser.cpp]: Right, I removed the value returned by the `update()` method.

File: src/hotspot/share/runtime/vmStructs.cpp (hunk 2 above)
Comment (fparain) [reply src/hotspot/share/runtime/vmStructs.cpp]: Not necessary, removed.

**Comment by fparain** [comment]
Hi Claes,

Thank you for your review, the new version should address the points you raised.

Fred

File: src/hotspot/share/oops/fieldInfo.hpp [hunk 3]
```diff
@@ -47,19 +47,21 @@ class FieldInfo {
...
   //   |---------high---------|---------low---------|
-  //    ..........................................00  - blank
+  //    ..........................................CO
+  //    ..........................................00  - non-contended field
+  //    [--contention_group--]....................10  - contended field with contention group
   //    [------------------offset----------------]01  - real field offset
-  //    ......................[-------type-------]10  - plain field with type
-  //    [--contention_group--][-------type-------]11  - contended field with type and contention group
+
+  // Bit O indicates if the packed field contains an offset (O=1) or not (O=1)
```
Comment (hseigel) [inline src/hotspot/share/oops/fieldInfo.hpp:61]: Hi Fred, should this comment say  "... or not (0=0)   ?

**Review by lfoltan — APPROVED** [review]

**Comment by lfoltan** [comment]
Looks good.

**Review by lfoltan — APPROVED** [review]
Looks good Fred.

File: src/hotspot/share/oops/fieldInfo.hpp (hunk 3 above)
Comment (fparain) [reply src/hotspot/share/oops/fieldInfo.hpp]: You're right. I've fixed the comment (and the line below which had the same issue).

Fred

**Review by hseigel — APPROVED** [review]
Looks good!

**Comment by fparain** [comment]
Claes, Lois, Harold,

Thank you for your reviews and feedback.

Fred

**Pull request closed without merging**